### PR TITLE
Bug 1829156 - Target SDK 34 in Focus.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1829156

### CODE REVIEW

**Comment by jonalmeida** [comment]
I think we need to target GV first before we do this in our applications, right?

File: focus-android/app/build.gradle [hunk 1]
```diff
@@ -24,7 +24,7 @@ android {
     defaultConfig {
         applicationId "org.mozilla"
         minSdkVersion config.minSdkVersion
-        targetSdkVersion config.targetSdkVersion
```
Comment (jonalmeida) [inline focus-android/app/build.gradle:27]: Assuming we're going to update this `config.targetSdkVersion` when we're done targetting AC/Fenix too?

**Comment by mcarare** [comment]
> I think we need to target GV first before we do this in our applications, right?

No.

**Comment by jonalmeida** [comment]
> > I think we need to target GV first before we do this in our applications, right?
> 
> No.

I recall now. We do need GV to target it first because it can lead to crashes where we may not be ready to handle.

I know this because I explicitly filed [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1743469#c0), with saying, "...and GV needs to do this first" so that we didn't get crashes by surprise as a result in the next version we had to target. 😄 

I can't find the fall out bugs from SDK 29 right now unfortunately, but I can try to look for them again if needed.

**Review by jonalmeida — APPROVED** [review]
Since we're looking to update all our dependencies to target 34 ASAP. Let's land this today and let us verify any crashes with our nightly user (singular 😄 ) and beta, where QA also does verification.

**Pull request merged**